--- a/src/freedreno/vulkan/meson.build
+++ b/src/freedreno/vulkan/meson.build
@@ -92,6 +92,10 @@ if freedreno_kmds.contains('kgsl')
   libtu_files += files('tu_knl_kgsl.cc')
 endif
 
+if host_machine.system() == 'android'
+  libtu_files += files('tu_atrace_compat.c')
+endif
+
 if freedreno_kmds.contains('msm')
   tu_flags += '-DTU_HAS_MSM'
   libtu_files += files('tu_knl_drm_msm.cc', 'tu_knl_drm.cc')
