diff --git i/CMakeLists.txt w/CMakeLists.txt
index c4c06a0..44fc35b 100644
--- i/CMakeLists.txt
+++ w/CMakeLists.txt
@@ -348,6 +348,7 @@ option_with_deps(XRT_BUILD_DRIVER_UVC "Enable userspace UVC driver" DEPENDS XRT_
 
 # Drivers to build (sorted)
 option_with_deps(XRT_BUILD_DRIVER_ANDROID "Enable Android sensors driver" DEPENDS ANDROID)
+option_with_deps(XRT_BUILD_DRIVER_PN2 "Enable Pico Neo 2 driver" DEPENDS ANDROID)
 option_with_deps(XRT_BUILD_DRIVER_ARDUINO "Enable Arduino input device with BLE" DEPENDS XRT_HAVE_DBUS)
 option_with_deps(XRT_BUILD_DRIVER_BLUBUR_S1 "Enable 3Glasses Blubur S1 device driver" DEPENDS XRT_HAVE_INTERNAL_HID)
 option_with_deps(XRT_BUILD_DRIVER_CONTACTGLOVE "Enable Melt Interface Technologies ContactGlove (CG2) device driver" DEPENDS XRT_HAVE_LINUX) # only Linux has serial right now
@@ -480,6 +481,7 @@ list(
 	"ILLIXR"
 	"NS"
 	"OHMD"
+	"PN2"
 	"OPENGLOVES"
 	"PSMV"
 	"PSSENSE"
diff --git i/src/xrt/drivers/CMakeLists.txt w/src/xrt/drivers/CMakeLists.txt
index 5ec9910..bd2fa48 100644
--- i/src/xrt/drivers/CMakeLists.txt
+++ w/src/xrt/drivers/CMakeLists.txt
@@ -526,6 +526,25 @@ if(XRT_BUILD_DRIVER_ANDROID)
 	list(APPEND ENABLED_DRIVERS android)
 endif()
 
+if(XRT_BUILD_DRIVER_PN2)
+	add_library(
+		drv_pn2 STATIC pn2/pn2_hmd.c pn2/pn2_hmd.h pn2/pn2_qvr.c pn2/pn2_qvr.h
+		pn2/pn2_ctrl.c pn2/pn2_ctrl.h pn2/ctrl_state.c pn2/ctrl_state.h
+		pn2/pn2_prober.c pn2/pn2_interface.h
+	)
+	target_link_libraries(
+		drv_pn2
+		PRIVATE
+			xrt-interfaces
+			aux_util
+			aux_os
+			aux_math
+			${ANDROID_LIBRARY}
+			${CMAKE_DL_LIBS}
+	)
+	list(APPEND ENABLED_HEADSET_DRIVERS pn2)
+endif()
+
 if(XRT_BUILD_DRIVER_ILLIXR)
 	add_library(
 		drv_illixr STATIC
diff --git i/src/xrt/targets/common/CMakeLists.txt w/src/xrt/targets/common/CMakeLists.txt
index f3f46c9..f8770cf 100644
--- i/src/xrt/targets/common/CMakeLists.txt
+++ w/src/xrt/targets/common/CMakeLists.txt
@@ -215,6 +215,10 @@ if(XRT_BUILD_DRIVER_ANDROID)
 	target_link_libraries(target_lists PRIVATE drv_android)
 endif()
 
+if(XRT_BUILD_DRIVER_PN2)
+	target_link_libraries(target_lists PRIVATE drv_pn2)
+endif()
+
 if(XRT_BUILD_DRIVER_ILLIXR)
 	target_link_libraries(target_lists PRIVATE drv_illixr)
 endif()
diff --git i/src/xrt/targets/common/target_lists.c w/src/xrt/targets/common/target_lists.c
index 78efb9b..cfd6aee 100644
--- i/src/xrt/targets/common/target_lists.c
+++ w/src/xrt/targets/common/target_lists.c
@@ -68,6 +68,10 @@
 #include "android/android_prober.h"
 #endif
 
+#ifdef XRT_BUILD_DRIVER_PN2
+#include "pn2/pn2_interface.h"
+#endif
+
 #ifdef XRT_BUILD_DRIVER_ILLIXR
 #include "illixr/illixr_interface.h"
 #endif
@@ -247,6 +251,11 @@ xrt_auto_prober_create_func_t target_auto_list[] = {
     oh_create_auto_prober,
 #endif
 
+#ifdef XRT_BUILD_DRIVER_PN2
+    // Before the generic Android sensors driver.
+    pn2_create_auto_prober,
+#endif
+
 #ifdef XRT_BUILD_DRIVER_ANDROID
     android_create_auto_prober,
 #endif
